pkgs/tools/networking/hysteria/default.nix

```
@@ -1,21 +1,32 @@
 { lib
 , fetchFromGitHub
-, buildGoModule
+, buildGo118Module
 }:
-buildGoModule rec {
+buildGo118Module rec {
```

```suggestion

buildGoModule rec {
```
In the meantime go 1.18 is the default.

---

It used to be like this, and here was the error that appeared before I changed `buildGoModule` into `buildGo118Module`. Note the penultimate line, I don't know if the version incompatible is the reason, but everything is solved after specifying the 118 version.
```
....
github.com/spf13/viper
github.com/mholt/acmez
github.com/lucas-clemente/quic-go/http3
github.com/HyNetwork/hysteria/pkg/transport
github.com/caddyserver/certmagic
github.com/HyNetwork/hysteria/pkg/core
github.com/prometheus/client_golang/prometheus/promhttp
# github.com/caddyserver/certmagic
../go/pkg/mod/github.com/caddyserver/certmagic@v0.17.1/certmagic.go:408:11: undefined: any
../go/pkg/mod/github.com/caddyserver/certmagic@v0.17.1/certmagic.go:428:13: undefined: any
../go/pkg/mod/github.com/caddyserver/certmagic@v0.17.1/config.go:68:66: undefined: any
../go/pkg/mod/github.com/caddyserver/certmagic@v0.17.1/config.go:1115:80: undefined: any
../go/pkg/mod/github.com/caddyserver/certmagic@v0.17.1/solvers.go:699:7: undefined: any
note: module requires Go 1.18
github.com/folbricht/routedns
....
```

Here is the [full log](https://gist.github.com/oluceps/763726f8f8deb17f4e79eabe83ddfb11)

---

buildGoModule is an alias to buildGo118Module https://github.com/NixOS/nixpkgs/blob/8a810b1a1611e7deaafc7b1bded2976b2a6ee5c2/pkgs/top-level/all-packages.nix#L22367

---

> buildGoModule is an alias to buildGo118Module https://github.com/NixOS/nixpkgs/blob/8a810b1a1611e7deaafc7b1bded2976b2a6ee5c2/pkgs/top-level/all-packages.nix#L22367

But the only modification that caused build failed is `buildGoModule`, that's weird

---

> > buildGoModule is an alias to buildGo118Module https://github.com/NixOS/nixpkgs/blob/8a810b1a1611e7deaafc7b1bded2976b2a6ee5c2/pkgs/top-level/all-packages.nix#L22367
> 
> But the only modification that caused build failed is `buildGoModule`, that's weird

And seems that some dependence reports require Go 1.18

---

> buildGoModule is an alias to buildGo118Module https://github.com/NixOS/nixpkgs/blob/8a810b1a1611e7deaafc7b1bded2976b2a6ee5c2/pkgs/top-level/all-packages.nix#L22367

Alright, it passed `nixpkgs-review rev HEAD`, works well with `buildGoModule`. And still failed to build with `nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'`